artcommon/artcommonlib/runtime.py

```diff
@@ -1,13 +1,67 @@
+import logging
+import traceback
+import warnings
 from abc import ABC, abstractmethod
 
+from artcommonlib import logutil
+from artcommonlib.konflux.konflux_db import KonfluxDb
+
+
 # an abstract class intended to be used by anything looking for config specific to a single
 # ocp-build-data group.
 
 
 class GroupRuntime(ABC):
+    def __init__(self, **kwargs):
+        self.debug = False
+        self.quiet = False
+        self.initialized = False
+        self._logger = None
+        self.konflux_db = None
+
+        for key, val in kwargs.items():
+            self.__dict__[key] = val
+
+    def initialize(self):
+        self.initialize_logging()
+        self.initialize_konxflux_db()
+
+    def initialize_logging(self):
+        if self.initialized or self._logger:
+            return
+
+        # Three flags control the output modes of the command:
+        # --verbose prints logs to CLI as well as to files
+        # --debug increases the log level to produce more detailed internal
+        #         behavior logging
+        # --quiet opposes both verbose and debug
+        if self.debug:
+            log_level = logging.DEBUG
+        elif self.quiet:
+            log_level = logging.WARN
+        else:
+            log_level = logging.INFO
+
+        logutil.setup_logging(log_level, self.debug_log_path)
+        self._logger = logging.getLogger('artcommonlib')
+
+    def initialize_konxflux_db(self):
+        try:
+            self.konflux_db = KonfluxDb()
+        except Exception as err:
+            self._logger.warning('Cannot connect to the Konflux DB: %s\n%s', str(err), traceback.format_exc())
```

Comment:
We should not forget to not allow production builds to go unrecorded. But local experiments should be fine (like not record development work elsewhere).
Fine to not take care right now of this, but is this tracked somewhere?

Reply:
Filed [ART-10665](https://issues.redhat.com/browse/ART-10665) for when we're ready to enforce the BigQuery DB setup in Doozer